consumers/campaigner/main.go

```diff
@@ -130,59 +130,70 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
 				logrus.WithFields(logrus.Fields{
 					"user_id":     msg.UserID,
 					"segment_ids": msg.SegmentIDs,
+					"campaign_id": msg.CampaignID,
 				}).WithError(err).Warn("unable to fetch subscribers.")
 				return nil
 			}
 			logrus.WithFields(logrus.Fields{
 				"user_id":     msg.UserID,
 				"segment_ids": msg.SegmentIDs,
+				"campaign_id": msg.CampaignID,
 			}).WithError(err).Error("unable to fetch subscribers.")
 			return nil
 		}
+
+		uid := ksuid.New()
```

Comment:
Is this invoked above the two loops?

Reply:
No only in the for loops

Reply:
We should maybe transfer this to be above the loops, and use Next() inside the loop.

Reply:
@FilipNikolovski i transferred it